cypress/tsconfig.json

```diff
@@ -1,9 +1,18 @@
 {
     "compilerOptions": {
         "target": "es2016",
-        "lib": ["es2020", "dom"],
+        "jsx": "react",
+        "lib": [
+            "es2020",
+            "dom",
+            "dom.iterable"
+        ],
         "types": ["cypress", "@percy/cypress"],
+        "resolveJsonModule": true,
+        "esModuleInterop": true,
```

Comment:
How can we resolve the remaining `TS1323: Dynamic imports are only supported when the '--module' flag is set to 'es2020', 'es2022', 'esnext', 'commonjs', 'amd', 'system', 'umd', 'node12', or 'nodenext'.` errors? [CI run](https://github.com/matrix-org/matrix-react-sdk/runs/6760841057?check_suite_focus=true)

It's unclear to me how we overcome this in the root config since it's also set to [`"module": "commonjs",`, `"target": "es2016",`](https://github.com/matrix-org/matrix-react-sdk/blob/7e244fc83303af5a4206785eaed316938e09a56d/tsconfig.json#L7-L9)

Reply:
> Given this is a draft, if you want support then I suggest https://matrix.to/#/#element-dev:matrix.org

Asked for help at https://matrix.to/#/!bEWtlqtDwCLFIAKAcv:matrix.org/$a0VvGZomu-9fCzmO1z-Y2_yJTvE8LuAf-WwYDOPvYew?via=matrix.org&via=element.io&via=mozilla.org

Reply:
Hm, does adding a `"module": "commonjs",` line to this file not work?

Reply:
> Hm, does adding a `"module": "commonjs",` line to this file not work?

I think that might have done the trick - let me make sure

Reply:
Let's hope :crossed_fingers: :D

Reply:
Thanks for suggestion @robintown! I assumed it would inherit like nested [ESLint configurations](https://eslint.org/docs/latest/user-guide/configuring/configuration-files#using-configuration-files) 😵

Thanks @SimonBrandner for moving everything forward here!